intranet/apps/polls/models.py

```diff
@@ -94,6 +94,30 @@ def get_users_voted(self):
                 users = list(q.get_users_voted())
         return users
 
+    def get_num_eligible_voters(self):
+        groups = get_user_model().objects.none()
+        if self.groups.exists():
+            q_objects = Q(name__in=[])
+            for g in self.groups.all():
+                q_objects.add(Q(name=g), Q.OR)
+            groups = DjangoGroup.objects.filter(q_objects)
+            users = set()
+            for g in groups:
```

Comment:
You don't need to create sets and use the `|` operator in order to merge different querysets together without duplicates. The SQL union operator (as exposed by Django's [union()](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#union)) returns a combined queryset that, by default, unless the keyword argument `all=True` is passed, only selects distinct values.

> The UNION operator selects only distinct values by default.
\- [Source](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#union)